ui/components/common/Validators.js

```diff
@@ -0,0 +1,43 @@
+const Validators = {
+    String: {
+        nonBlank: s => s
+                && typeof s === typeof ""
+                && s.length > 0,
+        minLength: l => s => Validators.String.nonBlank(s) && s.length >= l,
+        maxLength: l => s => Validators.String.nonBlank(s) && s.length < l
+    },
+    Number: {
+        natural: n => !Number.isNaN(n) && parseInt(n, 10) > 0
+    },
+    Port: {
+        valid: p => Validators.Number.natural(p) && p < 65536
```

Comment:
Should it not be a parameter?

Reply:
Ports can't be >65536, thats a constant

Reply:
So, should it not be better in a const var to make it clear? Following an example.

```javascript
const MAX_POSTHRES_PORT = 65536

{...}

valid: p => Validators.Number.natural(p) && p < MAX_POSTHRES_PORT
```